tests/consensus_tests/test_custom_sharding.py

```diff
@@ -112,4 +126,94 @@ def test_shard_consistency(tmp_path: pathlib.Path):
         }
     )
     assert_http_ok(r)
-    assert r.json()["result"]["count"] == 8
\ No newline at end of file
+    assert r.json()["result"]["count"] == 8
+
+    # Search points within the shard
+    r = requests.post(
+        f"{peer_api_uris[0]}/collections/{COLLECTION_NAME}/points/search",
+        json={
+            "vector": [0.29, 0.81, 0.75, 0.11],
+            "shard_key": "cats",
+            "limit": 10,
+            "with_payload": True,
+        }
+    )
+
+    assert_http_ok(r)
+    result = r.json()["result"]
+    assert len(result) == 4
+    for point in result:
+        assert point["payload"]["name"] in ["Barsik", "Murzik", "Vaska", "Chubais"]
+        assert point["shard_key"] == "cats"
+
+    # Search points within 2 shards
+    r = requests.post(
+        f"{peer_api_uris[0]}/collections/{COLLECTION_NAME}/points/search",
+        json={
+            "vector": [0.29, 0.81, 0.75, 0.11],
+            "shard_key": ["cats", "dogs"],
+            "limit": 10,
+            "with_payload": True,
+        }
+    )
+
+    assert_http_ok(r)
+    result = r.json()["result"]
+    assert len(result) == 8
+    for point in result:
+        assert point["shard_key"] in ["cats", "dogs"]
```

Comment:
Shall we also assert that we have results for at least both, and not just one of them?

Reply:
`assert len(result) == 8` should cover that